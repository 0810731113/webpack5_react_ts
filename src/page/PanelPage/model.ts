export interface QTResponse<T = {}> {
  actionname: string;
  response: {
    status: "ok" | "error";
    data: T;
  };
  arguments?: T;
}

export interface QTErrorMessage {
  errormessage: string;
}

export interface QTRequest {
  actionname: string;
  arguments?: {};
}

export interface LoginResponsePayload {
  accessToken: string;
  userId: string;
  userName: string;
}

export interface QTErrorData {
  errormessage: string;
  errornumber?: number;
}
