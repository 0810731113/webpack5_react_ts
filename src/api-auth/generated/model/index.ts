export * from './account-vo';
export * from './department';
export * from './enterprise-account';
export * from './enterprise-account-creation';
export * from './enterprise-account-creation-v2';
export * from './enterprise-account-query';
export * from './enterprise-info';
export * from './enterprise-member';
export * from './mobile-bind-account';
export * from './oauth2-access-token';
export * from './pagination-enterprise-member';
export * from './personal-account-creation';
export * from './response-account-vo';
export * from './response-enterprise-account';
export * from './response-enterprise-info';
export * from './response-list-department';
export * from './response-list-mobile-bind-account';
export * from './response-pagination-enterprise-member';
export * from './response-user-info';
export * from './response-user-info-with-token';
export * from './response-void';
export * from './responseboolean';
export * from './user-info';
export * from './user-info-with-token';
export * from './user-login';
