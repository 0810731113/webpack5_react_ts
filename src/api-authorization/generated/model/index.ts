export * from './resource-permission';
export * from './response-list-resource-permission';
export * from './response-list-user-roles';
export * from './response-liststring';
export * from './response-void';
export * from './user-role-creation';
export * from './user-roles';
