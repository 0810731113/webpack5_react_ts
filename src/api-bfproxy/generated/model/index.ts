export * from './bimface-upload-body';
export * from './bimface-upload-result';
export * from './condition';
export * from './data-set-integrate-bean';
export * from './data-set-integrate-request';
export * from './link';
export * from './model-and-view';
export * from './responsestring';
export * from './view';
