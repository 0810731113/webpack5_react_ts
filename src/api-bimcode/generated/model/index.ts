export * from './category';
export * from './category-property-ref';
export * from './data-code';
export * from './response-category';
export * from './response-category-property-ref';
export * from './response-list-data-code';
export * from './responseint';
export * from './type-table-translate';
