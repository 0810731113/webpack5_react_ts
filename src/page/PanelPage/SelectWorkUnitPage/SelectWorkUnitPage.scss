// .panel-body {
//   padding: 16px;

//   .title {
//     display: flex;
//     justify-content: space-between;
//     align-items: center;
//     margin: -8px -16px 14px;
//     padding: 0 16px;
//     .name {
//       font-size: 14px;
//     }

//     .action {
//       .anticon {
//         margin: 0 6px;

//         &:hover {
//           cursor: pointer;
//         }
//       }
//     }
//   }
// }
.empty-wrap {
  display: flex;
  align-items: center;
  width: 100%;
  height: 100%;
  justify-content: center;
}
