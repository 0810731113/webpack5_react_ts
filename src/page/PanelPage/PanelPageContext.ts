import { ResourcePermission } from "api-authorization/generated/model";
import { AxiosError } from "axios";
import { createContext } from "react";
import {
  DataSetVO,
  SpecialtyVO,
  SpecialtyVOTypeEnum,
  VersionVO,
} from "api/generated/model";
import { ReferedVersionVO } from "./ReferPage";

export enum ProjectContentEnum {
  Samples = "samples",
  Mine = "",
}

export interface PanelPageContext {
  userId: string | null;
  accessToken: string | null;
  workUnit: DataSetVO | null;
  workUnits: DataSetVO[] | null;
  version: VersionVO | null;
  projectId: string | null;
  specialtyType: SpecialtyVOTypeEnum;
  specialties?: SpecialtyVO[] | null;
  referingVersionIds: number[];
  referedVersions: ReferedVersionVO[];
  referWorkUnits: DataSetVO[] | null;
  workunitReadonly: boolean;
  showprocess: 0 | 1;
  logs: any[];
  resources: ResourcePermission[];
  projectContent: ProjectContentEnum | null;
  setSpecialties(value: SpecialtyVO[] | null | undefined): void;

  currentOperatingWorkUnitId: string | null;
  // 打开工作单元
  openingWorkUnit: boolean;
  setOpeningWorkUnit(value: boolean): void;

  // 保存工作单元
  savingWorkUnit: boolean;
  setSavingWorkUnit(value: boolean): void;

  // 提交工作单元
  commitingWorkUnit: boolean;
  setCommitingWorkUnit(value: boolean): void;
  setShowprocess(value: 0 | 1): void;

  setWorkUnit: (workUnit: DataSetVO | null) => void;
  setWorkUnits: (workUnits: DataSetVO[] | null) => void;
  setReferWorkUnits: (workUnits: DataSetVO[] | null) => void;
  setVersion: (version: VersionVO | null) => void;
  setCurrentOperationWorkUnitId: (id: string | null) => void;
  setProjectId: (id: string | null) => void;
  setReferingVersionIds: (ids: number[]) => void;
  setReferedVersions: (versions: ReferedVersionVO[]) => void;
  setWorkunitReadonly: (readonly: boolean) => void;
  pushLog: (log: any) => void;
  resetProject: () => void;
  setProjectContent: (content: ProjectContentEnum) => void;

  onResponseError: (error: AxiosError) => void;

  refreshCount: number;
}

export default createContext<PanelPageContext>({} as any);
