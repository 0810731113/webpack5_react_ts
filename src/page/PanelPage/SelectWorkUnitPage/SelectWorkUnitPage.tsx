import { InfoCircleFilled } from "@ant-design/icons";
import {
  Empty,
  message,
  Space,
  Tag,
  Button,
  Tooltip,
  Modal,
  Result,
} from "antd";
import { DataSetVO, Team, VersionVO } from "api/generated/model";
import { Select, SelectOption } from "component/Antd";
import PanelTabTitle from "component/PanelTabTitle";
import WorkUnitTree from "component/WorkUnitTree";
import { clone, first, orderBy, size, isNumber } from "lodash";
import React, { useContext, useEffect, Fragment } from "react";
import {
  teamService,
  workUnitService,
  specialtyService,
  versionService,
  authService,
  roleService,
} from "service";
import panelService from "service/panel.service";
import { useImmer } from "use-immer";
import "./SelectWorkUnitPage.scss";
import consts, { IS_MAC, TALKING_DATA_URL } from "consts";
import { mapWorkUnitBySpecialtyIdToTeam } from "function/workUnit.func";
import FakeProgress from "component/FakeProgress";
import qwebService from "service/qweb.service";
import { ProjectRole } from "service/role.service";
import { TooltipWrapper } from "component/wrapper/TooltipWrapper";
import { CheckPermissionPanel } from "component/CheckPermission/CheckPermissionPanel";
import { ResourcePermissionResourceEnum } from "api-authorization/generated/model";
import { publishEvent } from "function/stats.func";
import { specialtyTypeName } from "AppPanel";
import PanelPageContext from "../PanelPageContext";

const { PUBLIC_URL } = consts;
interface SelectWorkUnitPageProps {
  activeKey: string;
}

interface State {
  teams: Team[];
  versions: { [workUnitId: string]: VersionVO[] } | null;
  selectedVersions: { [workUnitId: string]: VersionVO | null };
  datasets: { [teamId: string]: DataSetVO[] } | null;
  openedTeamIds: string[];
  blocked: boolean;
}

export default function SelectWorkUnitPage({
  activeKey,
}: SelectWorkUnitPageProps) {
  const [
    { versions, selectedVersions, datasets, teams, openedTeamIds, blocked },
    updateState,
  ] = useImmer<State>({
    versions: null,
    datasets: null,
    selectedVersions: {},
    teams: [],
    openedTeamIds: [],
    blocked: false,
  });

  const {
    projectId,
    workUnit,
    workUnits,
    specialtyType,
    savingWorkUnit,
    currentOperatingWorkUnitId,
    commitingWorkUnit,
    openingWorkUnit,
    refreshCount,
    workunitReadonly,
    showprocess,
    setWorkUnit,
    setWorkUnits,
    setVersion,
    setProjectId,
    setCurrentOperationWorkUnitId,
    setOpeningWorkUnit,
    setSavingWorkUnit,
    setCommitingWorkUnit,
    setSpecialties,
    setWorkunitReadonly,
  } = useContext(PanelPageContext);

  useEffect(() => {
    const subscription = qwebService.subscribe((e) => {
      console.log("received qt event: ", e);
      // try {
      switch (e.CollaborationClientEvent.actionname) {
        case "saveworkunit":
          if (workUnit && workUnit.id) {
            setCurrentOperationWorkUnitId(workUnit.id);
            publishEvent("saveWorkUnit", ["工具"], {
              from: specialtyTypeName[specialtyType],
            });
          }
          setSavingWorkUnit(true);
          break;
        case "commitworkunit":
          if (workUnit && workUnit.id) {
            setCurrentOperationWorkUnitId(workUnit.id);
            publishEvent("commitWorkUnit", ["工具"], {
              from: specialtyTypeName[specialtyType],
            });
          }
          setCommitingWorkUnit(true);
          break;
        case "openinglatelyworkunit":
          const { projectid, datasetid } = e.CollaborationClientEvent.arguments;
          if (projectid) {
            setProjectId(projectid);
          }
          if (datasetid) {
            setCurrentOperationWorkUnitId(datasetid);
          }
          setOpeningWorkUnit(true);
          break;
        case "openedlatelyworkunit":
          const {
            projectid: openedProjectid,
            datasetid: openedDatasetid,
            status,
          } = e.CollaborationClientEvent.arguments;
          if (openedProjectid) {
            setProjectId(openedProjectid);
          }
          if (openedDatasetid) {
            setCurrentOperationWorkUnitId(openedDatasetid);
          }
          if (openedDatasetid && status !== "error") {
            versionService.getLatestVersion(openedDatasetid).then((version) => {
              if (
                version &&
                isNumber(version.version) &&
                version.verifyStatus !== "illegal"
              ) {
                const ds = workUnits?.find(
                  (dataset) => dataset.id === openedDatasetid,
                ) || { id: openedDatasetid };
                if (ds) {
                  setWorkUnit(ds);
                }
                setVersion(version);
                setWorkunitReadonly(status === "readonly");
              }
            });
          }
          setOpeningWorkUnit(false);
          setCurrentOperationWorkUnitId(null);
          break;

        default:
      }
    });
    return () => {
      subscription?.unsubscribe();
    };
  }, [workUnit]);

  const prepareData = async () => {
    if (projectId) {
      let currentSpecialtyId = null;

      // 拿专业
      if (specialtyType) {
        const specialties = await specialtyService
          .querySepcialty(projectId)
          .catch(() => {
            console.error("error in getting specialty");
            return null;
          });
        setSpecialties(specialties);

        const currentSpecialty = specialties?.find(
          (specialty) => specialty.type === specialtyType,
        );
        currentSpecialtyId = currentSpecialty?.id;
      }

      // workaround
      if (!currentSpecialtyId) {
        return;
      }

      // 拿团队
      const newTeams = await teamService.listMyTeam(projectId).catch(() => {
        console.error("error in getting teams");
        return null;
      });

      updateState((draft) => {
        draft.teams = newTeams!;
        draft.openedTeamIds = newTeams?.map((team) => team.id!) || [];
      });

      // 拿工作单元和版本
      if (newTeams instanceof Array && newTeams.length > 0) {
        const teamIds = newTeams.map((team) => team.id!);
        const newDatasets = await workUnitService
          .getBatchPanelWorkUnitsByTeamId(teamIds)
          .catch(() => {
            console.error("error in getting teams");
            return [];
          });
        // datasets = datasets.filter(
        //   (workUnit) => workUnit.ownerId === authService.getUserId(),
        // );
        const datasetsVO = mapWorkUnitBySpecialtyIdToTeam(
          newDatasets,
          currentSpecialtyId,
        );
        setWorkUnits(newDatasets);
        if (workUnit?.id) {
          const ds = workUnits?.find(
            (dataset) => dataset.id === currentOperatingWorkUnitId,
          );
          if (ds) {
            setWorkUnit({ ...ds });
          }
        }

        updateState((draft) => {
          draft.datasets = datasetsVO;
        });

        const dsIds = Object.values(datasetsVO)
          .flat()
          .map((dataset) => dataset.id!);

        if (dsIds instanceof Array && dsIds.length > 0) {
          const newVersions = await workUnitService
            .batchLoadWorkUnitsVersions(dsIds)
            .catch(() => {
              console.error("error in getting teams");
              return null;
            });

          updateState((draft) => {
            draft.versions = newVersions;
          });
        }
      }
    }
  };

  useEffect(() => {
    if (projectId) {
      roleService.getMyRole(projectId).then((roles) => {
        if (
          roles?.length === 1 &&
          roles[0] === ProjectRole.ProjectExternalUser
        ) {
          updateState((draft) => {
            draft.blocked = true;
          });
        } else {
          updateState((draft) => {
            draft.blocked = false;
          });
          prepareData();
        }
      });
    } else {
      updateState((draft) => {
        draft.teams = [];
      });
    }
  }, [projectId, specialtyType]);
  useEffect(() => {
    if (workUnit?.id) {
      const currentWorkunit = workUnits?.find(
        (unit) => unit.id === workUnit.id,
      );
      if (
        currentWorkunit &&
        JSON.stringify(currentWorkunit) !== JSON.stringify(workUnit)
      ) {
        setWorkUnit(currentWorkunit);
      }
    }
  }, [workUnits]);

  useEffect(() => {
    if (!savingWorkUnit) {
      prepareData();
    }
  }, [savingWorkUnit]);

  useEffect(() => {
    if (!commitingWorkUnit) {
      prepareData();
    }
  }, [commitingWorkUnit]);

  useEffect(() => {
    if (["1", "2"].includes(activeKey)) {
      prepareData();
    }
  }, [activeKey, refreshCount]);

  if (projectId === null) {
    return (
      <div className="empty-wrap">
        <Empty
          description={
            <Space direction="vertical">
              <span>请选择项目</span>
            </Space>
          }
        />
      </div>
    );
  }

  if (blocked) {
    return (
      <Empty
        description={
          <>
            <div>暂不支持外部成员查看项目</div>
            <div style={{ opacity: 0.21, fontSize: 12 }}>
              如需查看，请联系项目管理员修改权限
            </div>
          </>
        }
        image={`${PUBLIC_URL}/panel_blocked.svg`}
        style={{ marginTop: 48 }}
      />
    );
  }

  // if (!teams || teams.length === 0) {
  //   return (
  //     <Empty
  //       description={
  //         <Space direction="vertical">
  //           <span>暂无数据</span>
  //         </Space>
  //       }
  //     />
  //   );
  // }

  function saveWorkUnit() {
    if (workUnit && workUnit.id) {
      setCurrentOperationWorkUnitId(workUnit.id);
      publishEvent("saveWorkUnit", ["工具"], {
        from: specialtyTypeName[specialtyType],
      });
    }

    setSavingWorkUnit(true);
    panelService.saveWorkUnit(showprocess);
  }

  function commitWorkUnit() {
    if (workUnit && workUnit.id) {
      setCurrentOperationWorkUnitId(workUnit.id);
      publishEvent("commitWorkUnit", ["工具"], {
        from: specialtyTypeName[specialtyType],
      });
    }
    setCommitingWorkUnit(true);
    panelService.commitWorkUnit(showprocess);
  }

  function selectWorkUnit(ds: DataSetVO) {
    if (workUnit) {
      if (workUnit.id === ds.id) {
        message.info("当前工作单元已打开");
        return;
      }
      message.warn("请先关闭当前工作单元");
      return;
    }

    if (ds.id) {
      prepareData();
      setCurrentOperationWorkUnitId(ds.id);
      setOpeningWorkUnit(true);

      versionService.getLatestVersion(ds.id).then((version) => {
        if (
          version &&
          isNumber(version.version) &&
          version.verifyStatus !== "illegal"
        ) {
          panelService
            .openWorkUnit(ds.id!, version.version)
            .then((res: any) => {
              setWorkUnit(ds);
              setVersion(version);
              setWorkunitReadonly(!!res.readonly);
            })
            .catch((errornumber) => {
              if (errornumber.toString() === "3") {
                Modal.warning({
                  className: "clean-cache-modal",
                  title: "打开失败，需要清空缓存",
                  width: 280,
                  content: (
                    <div className="content-text">
                      <p>清空缓存方法：</p>
                      <p>
                        1.复制文件夹路径：%localappdata%\Glodon\
                        {specialtyType}
                      </p>
                      <p>
                        2.打开“计算机”，在路径处粘贴链接，找到“dbCache”文件夹
                      </p>
                      <p>3.关闭软件后，再删除“dbCache”文件夹</p>
                    </div>
                  ),
                  okButtonProps: { type: "default" },
                  okText: "确定",
                });
              }

              if (errornumber.toString() === "4") {
                Modal.error({
                  className: "clean-cache-modal",
                  title: "打开失败, 请升级软件",
                  content: (
                    <div className="content-text">
                      <p>此工作单元的数据已升级，请使用最新版本的工具端打开 </p>
                    </div>
                  ),
                  width: 280,
                  okButtonProps: { type: "default" },
                  okText: "确定",
                });
              }
            })
            .finally(() => {
              setOpeningWorkUnit(false);
              setCurrentOperationWorkUnitId(null);
            });
        } else {
          message.error(
            "版本有更新且数据异常，目前无法继续使用，请前往网页端恢复历史版本",
          );
        }
      });
    }
  }

  return (
    <div className="panel-body">
      <div className="title">
        <PanelTabTitle title="工作单元" tip="工作单元" />
      </div>
      <WorkUnitTree
        disabled={!!workUnit || openingWorkUnit}
        teams={teams}
        datasets={datasets}
        versions={versions}
        workUnitId={workUnit?.id}
        versionDisabled={() => true}
        nodeSelected={(dataSet) => dataSet.id === workUnit?.id}
        bottomActionRender={(dataSet, version) => {
          const isloading =
            currentOperatingWorkUnitId === dataSet.id &&
            (savingWorkUnit || commitingWorkUnit || openingWorkUnit);

          const illegal = version && version.verifyStatus === "illegal";

          return workUnit?.id === dataSet.id ? (
            <div>
              <CheckPermissionPanel
                resouseType={ResourcePermissionResourceEnum.PersonalDataset}
              >
                <TooltipWrapper
                  mouseEnterDelay={1}
                  when={(props) => props.disabled ?? false}
                  title="处于示例项目中无该功能权限"
                  defaultTitle="将工作单元保存至“广联达协同设计平台-个人设计"
                >
                  <Button
                    // type="link"
                    // shape="circle"
                    onClick={saveWorkUnit}
                    disabled={
                      workunitReadonly || !workUnit || isloading || illegal
                    }
                    size="small"
                  >
                    保存
                  </Button>
                </TooltipWrapper>
              </CheckPermissionPanel>

              <CheckPermissionPanel
                resouseType={ResourcePermissionResourceEnum.PersonalDataset}
              >
                <TooltipWrapper
                  mouseEnterDelay={1}
                  when={(props) => props.disabled ?? false}
                  title="处于示例项目中无该功能权限"
                  placement="bottomRight"
                  defaultTitle="将工作单元提交至“广联达协同设计平台-团队协同”"
                >
                  <Button
                    // type="link"
                    // shape="circle"
                    onClick={commitWorkUnit}
                    disabled={
                      workunitReadonly || !workUnit || isloading || illegal
                    }
                    size="small"
                  >
                    提交
                  </Button>
                </TooltipWrapper>
              </CheckPermissionPanel>
            </div>
          ) : (
            false
          );
        }}
        actionRender={(dataSet, version) => {
          const isloading =
            currentOperatingWorkUnitId === dataSet.id &&
            (showprocess
              ? openingWorkUnit
              : savingWorkUnit || commitingWorkUnit || openingWorkUnit);

          if (version && version.verifyStatus === "illegal") {
            return (
              <Tooltip
                trigger="hover"
                mouseEnterDelay={1}
                placement="bottom"
                title="非常抱歉，该版本数据似乎出现了问题，目前无法继续使用，请前往网页端恢复历史版本"
              >
                <InfoCircleFilled style={{ color: "#faad14" }} />
              </Tooltip>
            );
          }

          return isloading ? (
            <FakeProgress
              style={{ marginLeft: 9 }}
              completed={false}
              showInfo={false}
              width={20}
              onCompleted={() => {
                console.log("completed");
              }}
            />
          ) : (
            <Button
              size="small"
              style={{ fontSize: 12, padding: "0px 4px", minWidth: "auto" }}
              disabled={!!workUnit || openingWorkUnit}
              onClick={() => selectWorkUnit(dataSet)}
            >
              打开
            </Button>
          );
        }}
      />
    </div>
  );
}
