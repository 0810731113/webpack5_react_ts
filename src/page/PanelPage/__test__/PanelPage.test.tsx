import React, { PropsWithChildren } from "react";
import pretty from "pretty";
import { render, screen } from "@testing-library/react";
import { act } from "react-dom/test-utils";
import { render as renderHtml, unmountComponentAtNode } from "react-dom";
import { isPanel } from "consts";

import { ResourcePermissionResourceEnum } from "api-authorization/generated/model";
import { roleService } from "service";

import WorkUnitTree from "component/WorkUnitTree";
import PanelPageContext from "../PanelPageContext";
import SelectWorkUnitPage from "../SelectWorkUnitPage/SelectWorkUnitPage";

let container: HTMLDivElement;

beforeEach(() => {
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  if (container) {
    unmountComponentAtNode(container);
    container.remove();
  }
  jest.clearAllMocks();
});

jest.mock("service");
jest.mock("consts", () => ({
  isPanel: true,
  PUBLIC_URL: "/web",
}));

jest.mock("component/CheckPermission/CheckPermissionPlatform");

test("external user", async () => {
  const defaultContext = {
    userId: "6742693964083531997",
    accessToken: "cn-e5a6e077-8602-4f18-ad55-276e099f88e3",
    projectId: "3f9cbef2-e240-4797-97ab-2a29f9224f8c",
  };

  renderHtml(
    <PanelPageContext.Provider value={defaultContext as any}>
      <SelectWorkUnitPage activeKey="1" />
    </PanelPageContext.Provider>,

    container,
  );

  await act(async () => {
    await (roleService.getMyRole);
  });

  expect(pretty(container?.innerHTML)).toMatchSnapshot();
});

test("test open workunit", async () => {
  const defaultContext = {
    userId: "6742693964083531997",
    accessToken: "cn-e5a6e077-8602-4f18-ad55-276e099f88e3",
    projectId: "3f9cbef2-e240-4797-97ab-2a29f9224f8c",
    // workUnit: null,
    // workUnits: [],
    // version: null,
    // specialtyType: SpecialtyVOTypeEnum.GST,
    // referingVersionIds: [],
    // referedVersions: [],
    // referWorkUnits: [],
    // workunitReadonly: false,
    // showprocess: 0,
    // logs: [],
    // resources: [],
    // projectContent: null,
    // // setSpecialties:(x)=>void,
    // // setOpeningWorkUnit:(x)=>void,
  };

  const datasets = {
    "f84e38f9-c4f1-4263-97ca-1b8655e6b935": [
      {
        codeVersion: 1,
        description: "",
        folderId: "37e957e5-34bd-42e1-8d74-8a5ff7e2d5c0",
        id: "a0c3cb00-4656-447b-b168-b3beb8dd3cf1",
        name: "ds001",
        nameSpace: "",
        ownerId: "6742378634747412834",
        projectId: undefined,
        schemaDefinition: undefined,
        specialtyId: "9ab40471-5f4e-4d27-95d3-a4b40291eec8",
        teamId: "f84e38f9-c4f1-4263-97ca-1b8655e6b935",
        tipVersion: 4,
        type: "workunit",
      },
    ],
  };

  const versions = {
    "a0c3cb00-4656-447b-b168-b3beb8dd3cf1": [
      {
        creationTime: "2021-03-05T12:09:28.000+0000",
        dataSetId: "a0c3cb00-4656-447b-b168-b3beb8dd3cf1",
        dataSetSourceFile: "",
        id: 41973,
        metaInfo: `{"SpaceConfigVersion":"1"}`,
        status: "WIP",
        suiteCount: 3,
        verifyStatus: "Unverified",
        version: 3,
        viewingInfo: `{"jobToken":"991e63d1-28d1-4708-8d5d-af19eb702bee","fileId":2074152479410720,"status":"success"}`,
        xtoken: "eb9724c7-6fce-450e-8235-6f57a3658642",
      },
      {
        creationTime: "2021-03-05T11:15:56.000+0000",
        dataSetId: "a0c3cb00-4656-447b-b168-b3beb8dd3cf1",
        dataSetSourceFile: "",
        id: 41970,
        status: "WIP",
        suiteCount: 0,
        verifyStatus: "Unverified",
        version: 0,
      },
      {
        creationTime: "2021-03-05T11:17:01.000+0000",
        dataSetId: "a0c3cb00-4656-447b-b168-b3beb8dd3cf1",
        dataSetSourceFile: "",
        id: 41971,
        metaInfo: `{"SpaceConfigVersion":"1"}`,
        status: "WIP",
        suiteCount: 2,
        verifyStatus: "Unverified",
        version: 1,
        viewingInfo: `{"jobToken":"e1c4dc2f-d5b3-494b-94a5-4aa7aa1c3e18","fileId":2074126702414144,"status":"success"}`,
        xtoken: "e71684d0-da77-4f44-8549-9562ad0bf9a6",
      },
      {
        creationTime: "2021-03-05T11:18:01.000+0000",
        dataSetId: "a0c3cb00-4656-447b-b168-b3beb8dd3cf1",
        dataSetSourceFile: "",
        id: 41972,
        metaInfo: `{"SpaceConfigVersion":"1"}`,
        status: "WIP",
        suiteCount: 2,
        verifyStatus: "Unverified",
        version: 2,
        viewingInfo: `{"jobToken":"f188cb65-3194-438b-96af-6c364a563975","fileId":2074127184529952,"status":"success"}`,
        xtoken: "e71684d0-da77-4f44-8549-9562ad0bf9a6",
      },
    ],
  };

  renderHtml(
    <PanelPageContext.Provider value={defaultContext as any}>
      <WorkUnitTree
        disabled={false}
        teams={[
          {
            creationTime: "2021-03-05T11:15:47.000+0000",
            description: "",
            id: "f84e38f9-c4f1-4263-97ca-1b8655e6b935",
            linkedFolderId: "37e957e5-34bd-42e1-8d74-8a5ff7e2d5c0",
            name: "GAP",
            ownerId: "6742378634747412834",
            projectId: "3f9cbef2-e240-4797-97ab-2a29f9224f8c",
            status: undefined,
          },
        ]}
        datasets={datasets}
        versions={versions}
      />
    </PanelPageContext.Provider>,

    container,
  );

  expect(pretty(container?.innerHTML)).toMatchSnapshot();
});
