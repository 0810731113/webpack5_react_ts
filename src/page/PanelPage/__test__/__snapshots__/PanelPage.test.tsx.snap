// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`external user 1`] = `
"<div class=\\"ant-empty\\" style=\\"margin-top: 48px;\\">
  <div class=\\"ant-empty-image\\"><img alt=\\"empty\\" src=\\"/web/panel_blocked.svg\\"></div>
  <div class=\\"ant-empty-description\\">
    <div>暂不支持外部成员查看项目</div>
    <div style=\\"opacity: 0.21; font-size: 12px;\\">如需查看，请联系项目管理员修改权限</div>
  </div>
</div>"
`;

exports[`test open workunit 1`] = `
"<div class=\\"work-unit-tree  \\">
  <div class=\\"team-block \\">
    <div class=\\"team-name\\"><span role=\\"img\\" aria-label=\\"caret-down\\" tabindex=\\"-1\\" class=\\"anticon anticon-caret-down\\"><svg viewBox=\\"0 0 1024 1024\\" focusable=\\"false\\" data-icon=\\"caret-down\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path d=\\"M840.4 300H183.6c-19.7 0-30.7 20.8-18.5 35l328.4 380.8c9.4 10.9 27.5 10.9 37 0L858.9 335c12.2-14.2 1.2-35-18.5-35z\\"></path></svg></span><span>GAP</span></div>
    <div class=\\"team-content\\">
      <div class=\\"node \\"><span class=\\"ant-typography ant-typography-ellipsis\\" style=\\"width: 120px; margin: 0px 0px -4px 0px; overflow: scroll;\\">ds001</span>
        <div class=\\"node-version\\">
          <div class=\\"ant-select ant-select-sm ant-select-single ant-select-show-arrow\\" style=\\"width: 70px;\\">
            <div class=\\"ant-select-selector\\"><span class=\\"ant-select-selection-search\\"><input autocomplete=\\"off\\" type=\\"search\\" class=\\"ant-select-selection-search-input\\" style=\\"opacity: 0;\\" role=\\"combobox\\" aria-haspopup=\\"listbox\\" aria-owns=\\"rc_select_TEST_OR_SSR_list\\" aria-autocomplete=\\"list\\" aria-controls=\\"rc_select_TEST_OR_SSR_list\\" aria-activedescendant=\\"rc_select_TEST_OR_SSR_list_0\\" readonly=\\"\\" unselectable=\\"on\\" value=\\"\\" id=\\"rc_select_TEST_OR_SSR\\"></span><span class=\\"ant-select-selection-item\\">V3</span></div><span class=\\"ant-select-arrow\\" style=\\"user-select: none;\\" unselectable=\\"on\\" aria-hidden=\\"true\\"><span role=\\"img\\" aria-label=\\"down\\" class=\\"anticon anticon-down ant-select-suffix\\"><svg viewBox=\\"64 64 896 896\\" focusable=\\"false\\" data-icon=\\"down\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path d=\\"M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z\\"></path></svg></span></span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>"
`;
